### DISCUSSION TITLE
Reproductibility - Is there a way to specify the seed number as in Optuna?

### DISCUSSION THREAD

**Discussion by jessssschiu** [Tunny-gh/Tunny_in_Open_Sea#169] [General]
I am using the TPE, GP and NSGA-II samplers.

In Optuna, one can specify the seed number for the random number generator, is it possible with Tunny?

Otherwise, is there a way to reproduce the results with Tunny?

**Comment by hrntsm** [comment]
Hi @jessssschiu 

Tunny can also set the seed value. However, it is not possible from the UI.
Tunny has a settings file, which can be manipulated to set the seed values.
The Tunny UI reads and writes each value set in this file and reflects the settings in the optimisation.

The settings file is created in the same location as the .gha file, as follows.

<img width="698" alt="Screenshot 2023-04-04 at 11 40 19" src="https://user-images.githubusercontent.com/23289252/229672799-3d7e8939-23f4-407c-aed8-3834373cc745.png">


When opened in a text editor, it looks like this In each algorithm section, there is a seed entry. If it is "null", it is an automatic setting, but you can specify a seed value by entering an integer of your choice.
![Screenshot 2023-04-04 at 11 46 58](https://user-images.githubusercontent.com/23289252/229673684-b2d67c70-c990-4f89-b762-0858d1691a22.png)

Tunny loads this file when the Tunny UI is launched on Grasshopper and saves the settings in the UI to this file when the Tunny UI is closed.
So it is recommended to close the Tunny UI when editing the file directly.

**Reply by jessssschiu** [reply]
Thanks for the quick answer. It works! :)

**Reply by jessssschiu** [reply]
Seems that the possibility of changing the Settings.JSON has been removed in the latest versions. Is it still possible to set the seed number for all the random processes?

**Reply by hrntsm** [reply]
Hi @jessssschiu ,

I've changed the settings file from the latest version to the following location.
[UserProfile]/.tunny_env (e.g. "C:\Users\hrntsm \ .tunny_env\settings.json")

If you edit it there as you have done so far, it will be reflected.

**Reply by jessssschiu** [reply]
Thanks! I managed to find the settings file, but after I have made and saved changes in the JSON file, I failed to open the Tunny popup window again. Could this be something related to the settings.json, or another issue?

**Reply by hrntsm** [reply]
I tried it with the current development version and didn't have that problem.
I've uploaded the development version so you can try it.

https://drive.google.com/file/d/1lbVNsRcENyvYdkiouMUHCiFTaFFyZvCz/view?usp=drive_link

**Reply by jessssschiu** [reply]
Thanks again for the quick reply. I reinstalled Tunny 0.9 and it worked again :)
